shared/fixes/bash/accounts_authorized_local_users_sidadm_orasid.sh

```diff
@@ -25,23 +40,22 @@ if [ -d "/sapmnt" ] ; then
 
 	# if brspace exist in any of the above directory of a SID, add orasid to the userlist 
 	for path_to_brspace in $path_to_brspace_list ; do
-        	SID=${path_to_brspace:8:3}
-        	userlist="$userlist|ora$(echo "$SID" | sed -e 's/\(.*\)/\L\1/')"
+		SID=${path_to_brspace:8:3}
+		userlist="$userlist|ora$(echo "$SID" | sed -e 's/\(.*\)/\L\1/')"
 	done
 
 	# if owner of any brspace file is oracle, add oracle to the userlist
-	oracle=false
-	for path_to_brspace in $path_to_brspace_list ; do
-        	if [ $(ls -ld $path_to_brspace | awk '{print $3}') = "oracle" ]; then
-                	oracle=true
-        	fi
-	done
-	if test "$oracle" = true ; then
-        	userlist="$userlist|oracle"
-	fi
+	if test "$(is_owner_oracle "${path_to_brspace_list[@]}")" = 1 ; then userlist="$userlist|oracle" ; fi
+fi
+
+# if owner of any /oracle/SID directory is oracle, add oracle to the userlist
+# the user oracle could be added twice in the userlist, but it is harmlos to the final result
+if [ -d "/oracle" ] ; then
+	path_oracle_SID_list=$(find /oracle/ -regex "^$oracle_SID_stem$")
+	if test "$(is_owner_oracle "${path_oracle_SID_list[@]}")" = 1 ; then userlist="$userlist|oracle" ; fi
```

Comment:
I see that we have a duplicated line here, what about extracting it to a separate function that would accept array as a parameter?

Reply:
I was trying to avoid modify the value of global variable inside a function, which looks more dangerous than repeated code lines to me. Anyway, I moved the repeated lines to the function above, you may choose which one you prefer :-)  Thanks.